### ISSUE TITLE
how to change a single layer learning rate when fine tune ?

### ISSUE THREAD

**Issue by maxpy** [dmlc/mxnet#492]
Just like  blobs_lr: 0  param , in  Caffe  .prototxt   setting

**Comment by winstywang** [comment]
You may need to modify corresponding python file. If your goal is to freeze some early layers from finetuning, you can you the BlockGradOp.

BTW, it is recommended to use English here, since the users of mxnet are not only limited to Chinese.

**Comment by piiswrong** [comment]
I have a patch for this but haven't merged it in. I'll paste it here as an example:
index ccfb99e..6bb5411 100644
--- a/python/mxnet/optimizer.py
+++ b/python/mxnet/optimizer.py
@@ -66,7 +66,7 @@ class Optimizer(object):
         """Create additional optimizer state such as momentum.
         override in implementations."""
 
-    def update(self, index, weight, grad, state):
+    def update(self, index, weight, grad, state, lr_mult=1.0):
         """Update the parameters. override in implementations"""
 
 #convenience wrapper for Optimizer.Register
@@ -120,7 +120,7 @@ class SGD(Optimizer):
         else:
             return zeros(weight.shape, weight.context)
 
-    def update(self, index, weight, grad, state):
+    def update(self, index, weight, grad, state, lr_mult=1.0):
         """Update the parameters.
 
         Parameters
@@ -144,6 +144,7 @@ class SGD(Optimizer):
             lr = self.lr_scheduler(self.epoch)
         else:
             lr = self.lr
+        lr *= lr_mult
 
         grad = grad * self.rescale_grad
         if self.clip_gradient != None:
@@ -191,9 +192,9 @@ def get_updater(optimizer):
          The clossure of the updater
     """
     states = dict()
-    def updater(index, grad, weight):
+    def updater(index, grad, weight, lr_mult=1.0):
         """updater for kvstore"""
         if index not in states:
             states[index] = optimizer.create_state(index, weight)
-        optimizer.update(index, weight, grad, states[index])
+        optimizer.update(index, weight, grad, states[index], lr_mult=1.0)
     return updater

**Comment by maxpy** [comment]
merge  the optimize.py ,
 and change model.py  line 337 to updater(index*num_device+k, g, w , 0)   in 
is that right?

**Comment by piiswrong** [comment]
That's the basic idea
We need a proper solution but that might take some time. You are welcome to submit PR if you are interested in contributing.

**Comment by apark263** [comment]
so for the graph showing training evolution here:

https://mxnet.readthedocs.org/en/latest/tutorial/imagenet_full.html

are you just checkpointing and then manually changing your learning rate at each transition point?
